cmd/image-builder/main.go

```diff
@@ -69,6 +104,17 @@ operating sytsems like centos and RHEL with easy customizations support.`,
 	listImagesCmd.Flags().String("output", "", "Output in a specific format (text, json)")
 	rootCmd.AddCommand(listImagesCmd)
 
+	manifestCmd := &cobra.Command{
+		Use:          "manifest <distro> <image-type> [<arch>]",
```

Comment:
Would it be nicer if the architecture was a flag option (`--arch`)?
That way we could make blueprint an optional positional argument, which I think would be a more common usage.  Switching the arch would probably be less common.

Reply:
Thanks, that is an interesting idea. The reason for current layout is that it's designed to be copy/paste friendly from `list-iamges`. I.e. 
```console
$ image-builder list-images
...
rhel-10.0 type:oci arch:x86_64
$ image-builder manifest rhel-10.0 type:oci arch:x86_64
...
```
so we could tweak the list output to
```console
$ image-builder list-images
...
rhel-10.0 type:vhd --arch x86_64
rhel-10.0 type:vmdk --arch x86_64
rhel-10.0 type:oci --arch x86_64
$ image-builder manifest rhel-10.0 type:oci --arch x86_64
...
```
(which looks a bit uneven but not too bad and for scripts we provide --json output).

Or we could drop this property (copy/pastable). Or we make the third argument fuzzy, i.e. if it starts with "arch:" we consider it an architecture and otherwise its a blueprint. (or something else I forgot?)

WDYT?

Reply:
> While that makes typing out a manifest or build command by hand a tiny bit more tedious, it also makes it nice that the args aren't positional anymore and opens up the way to make distro optional too.

One issue with this is that we're inventing a new convention for non-positional arguments with `option:value` instead of the standard `--option=value` ...

Reply:
[..]
> > Or we could drop this property (copy/pastable).
> 
> It's a nice property but yes I don't think it's important enough to make the output of another command so unusual.
[..]

Let's do that then, it's the most simple and straightforward change. I am a little bit sad about it because now:
```console
$ image-builder manifest distro:rhel-10.0 type:oci arch:x86_64
error: cannot find blueprint "arch:x86_64"

$ image-builder build distro:rhel-10.0 type:oci arch:x86_64
error: cannot find blueprint "arch:x86_64"
```

We could change the default for `image-builder-cli list-image` to only show same arch images by default and hint at something like "--all" to see all, then we would have preserved the copy/paste property again (at least by default) and it's nicer for the "build" case. Open for ideas of course but that seems a reasonable path forward(?).

Reply:
Fwiw, I have no strong opinion, I like the copy/paste property of the original design and "bib" also rquires an argument (currently) to pass a blueprint. Going back to the previous way is as easy as removing the last commit :) But I'm more interested in a way forward right now than which way it is :) (I would love to get this in before EOY)

Reply:
This had some more (offline) discussion with Ondrej and Achilleas and the latest agreement is:
```
This commit tweaks the argument handling based on the suggestion
by Achilleas and Ondrej (thanks!). Now it takes

$ image-builder manifest qcow2 ./path/to/blueprint
...
$ image-builder manifest --arch s390x --distro centos-9 qcow2
...

If no arch is specified the hostname arch is used. If no distro
is specified in either the blueprint or the commandline it is
auto-detected based on the host.

Note that if the distro from the comandline and the blueprint
diverge an error is raised. We can relax this rule and just
add precedence, e.g. commandline always overrides the blueprint
but ideally we would have a clear use-case here so we start
conservative and can always relax this rule later (the inverse
is much harder).

This means it is no longer copy/paste friendly from `list-images`
by default but instead we can provide a new
`list-images --output=shell` option that outputs in exactly the
format that `image-builder [manifest|build]` need.
```